resteasy-core-spi/src/main/java/org/jboss/resteasy/spi/util/Types.java

```diff
@@ -810,7 +810,7 @@ public static boolean hasPostConstruct(Class<?> clazz)
 
    public static boolean hasPostConstruct(Class<?> clazz, Function<Method, Boolean> validateParameterCount)
    {
-      for (Method m : clazz.getMethods())
+      for (Method m : clazz.getDeclaredMethods())

```

Comment:
Hmmm. You're right, getMethods() returns only public methods. On the other hand, getDeclaredMethods() doesn't look at inherited methods. Probably needs to call getDeclaredMethods() while recursively visiting all superclasses. Thank you.

Reply:
Thanks @ronsigal I didn't think of that.
Added a recursive check for superclasses, if it looks good, I'll squash the two commits

Reply:
So far, so good, @spyrkob. Thanks.

There's also the alternative signatures for Interceptors, but I still have to look at the EJB case.